src/Symfony/Component/DependencyInjection/Loader/PhpFileLoader.php

```diff
@@ -102,7 +102,7 @@ private function executeCallback(callable $callback, ContainerConfigurator $cont
         foreach ($parameters as $parameter) {
             $reflectionType = $parameter->getType();
             if (!$reflectionType instanceof \ReflectionNamedType) {
-                throw new \InvalidArgumentException(sprintf('Could not resolve argument "$%s" for "%s".', $parameter->getName(), $path));
+                throw new \InvalidArgumentException(sprintf('Could not resolve argument "$%s" for "%s". You must typehint it (for example with "%s" or "%s").', $parameter->getName(), $path, ContainerConfigurator::class, ContainerBuilder::class));
```

Comment:
the message is not clear. there are more allowed types

Reply:
That's why I said "for example". Other class are for more advanced usage,isn't ? And the message would be too long IMHO

Reply:
I'd propose not adding any suggestion but just telling that the argument is missing a type hint.

`'Could not resolve argument "$%s" for "%s" as it misses a type hint.`

Reply:
So people must open the class to see what they must use instead of throwing a better message? Why do we want to reduce the DX?